contracts/src/core/libraries/v3/config-registry/ConfigRegistryLib.sol

```diff
@@ -5,84 +5,168 @@ import {ConfigRegistryStorage as S} from "src/core/libraries/v3/config-registry/
 import {ConfigRegistryTypes as T} from "src/core/libraries/v3/config-registry/ConfigRegistryTypes.sol";
 
 library ConfigRegistryLib {
-    event ConfigBytes32Set(bytes32 nameDigest, uint256 activationKey, bytes32 value);
     event ConfigBytesSet(bytes32 nameDigest, uint256 activationKey, bytes value);
 
+    /// @notice Thrown when attempting to retrieve a configuration by an unregistered name digest
+    /// @param nameDigest The unregistered name digest
     error NameDigestNotRegistered(bytes32 nameDigest);
+
+    /// @notice Thrown when trying to add a configuration with an activation key that is not strictly increasing
+    /// @param previousActivationKey The last activation key for this configuration
+    /// @param newActivationKey The activation key being added (must be > previousActivationKey)
     error NotIncreasingActivationKey(uint256 previousActivationKey, uint256 newActivationKey);
 
+    /// @notice Thrown when adding the first block number configuration with an activation block in the past
+    /// @param currBlock The current block number
+    /// @param abn The activation block number being added (must be >= currBlock)
+    error BlockNumberActivationKeyInPast(uint256 currBlock, uint256 abn);
+
+    /// @notice Thrown when adding the first timestamp configuration with an activation timestamp in the past
+    /// @param currTS The current timestamp
+    /// @param ats The activation timestamp being added (must be >= currTS)
+    error TimeStampActivationKeyInPast(uint256 currTS, uint256 ats);
```

Comment:
How closely do the block times actually match real-world times? Is is technically possible for block timestamps to drift from real time?

The block-number assertion clearly makes sense, but I lean toward saying we _shouldn't_ be making this timestamp assertion.

Reply:
hmm - iiuc there's only around 15 seconds of allowed drift that a block miner is allowed wrt current time: https://metaschool.so/articles/what-is-block-timestamp-in-solidity

Reply:
from where I got 15 seconds from: https://owasp.org/www-project-smart-contract-top-10/2023/en/src/SC03-timestamp-dependence.html